tools/Makefile.unix

```diff
@@ -501,6 +501,23 @@ ifeq ($(CONFIG_UBOOT_UIMAGE),y)
 		cp -f uImage /tftpboot/uImage; \
 	fi
 endif
+ifeq ($(CONFIG_ESP32_BINARY),y)
+	@echo "MKIMAGE: ESP32 binary"
+	$(Q) if ! esptool.py version | grep "v2[.]"; then \
```

Comment:
Could we let QEMU directly load ELF image instead binary? So we don't include the speicial version of gcc from silicon vendor.

Reply:
I tried without success, but maybe there is some way, since I am not familiar with it, sorry.

Reply:
I am thinking that a more systematic solution is needed both for CXD56xx and ESP32 and for any future platforms that require custom binary generation after the final linking step.  I have started some more general discussion in #442 

I am thinking that perhaps we should support some Makefiles and tools in directories under tools/.  Then perhaps we could do a wildcard check for:

(TOPDIR)/tools/$(CONFIG_ARCH)/Makefile

And if it exists, then run it.  So in this case, it would check for tools/esp32/Makefile and would execture tools/esp32/Makefile to generate the binary using the changes in this PR.

This is more effort, but would serve the needs of all architectures, existing and future.

Reply:
Loading an ELF file directly should be possible, provided that the ELF file includes the necessary CPU startup code and exception vectors. This doesn't happen when you build an ESP-IDF application — currently it always relies on the ROM code to provide the startup code and the exception vectors. However i think that NuttX includes such an option: https://github.com/apache/incubator-nuttx/blob/119a38ce10d8be96b8373b5a19145eda231edd93/boards/xtensa/esp32/esp32-core/README.txt#L491-L499 (i haven't tried it myself).

Reply:
Thanks for the info, I will try, it could be a simpler way than the current approach.